internal/cluster/distribution/pke/service.go

```diff
@@ -0,0 +1,215 @@
+// limitations under the License.
+
+package pke
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+
+	"github.com/banzaicloud/pipeline/internal/cluster"
+)
+
+type sentinel string
+
+func (e sentinel) Error() string {
+	return string(e)
+}
+
+func (e sentinel) ServiceError() bool {
+	return true
+}
+
+const notPipelineEnterpriseError = sentinel("pke nodepool update is supported only in Pipeline Enterprise")
+
+// +testify:mock
+
+// Service provides an interface to PKE AWS clusters.
+type Service interface {
+	// UpdateCluster updates a cluster.
+	//
+	// This method accepts a partial body representation.
+	UpdateCluster(ctx context.Context, clusterID uint, clusterUpdate ClusterUpdate) error
+
+	// UpdateNodePool updates an existing node pool in a cluster.
+	//
+	// This method accepts a partial body representation.
+	UpdateNodePool(ctx context.Context, clusterID uint, nodePoolName string, nodePoolUpdate NodePoolUpdate) (string, error)
+
+	// ListNodePools lists node pools from a cluster.
+	ListNodePools(ctx context.Context, clusterID uint) ([]NodePool, error)
+}
+
+// ClusterUpdate describes a cluster update request.
+//
+// A cluster update contains a partial representation of the cluster resource,
+// updating only the changed values.
+type ClusterUpdate struct {
+	Version string `mapstructure:"version"`
+}
+
+// NodePoolUpdate describes a node pool update request.
+//
+// A node pool update contains a partial representation of the node pool resource,
+// updating only the changed values.
+type NodePoolUpdate struct {
+	Image string `mapstructure:"image"`
+
+	Version string `mapstructure:"version"`
+
+	Options NodePoolUpdateOptions `mapstructure:"options"`
+}
+
+type NodePoolUpdateOptions struct {
+
+	// Maximum number of extra nodes that can be created during the update.
+	MaxSurge int `mapstructure:"maxSurge"`
+
+	// Maximum number of nodes that can be updated simultaneously.
+	MaxBatchSize int `mapstructure:"maxBatchSize"`
+
+	// Maximum number of nodes that can be unavailable during the update.
+	MaxUnavailable int `mapstructure:"maxUnavailable"`
+
+	// Kubernetes node drain specific options.
+	Drain NodePoolUpdateDrainOptions `mapstructure:"drain"`
+}
+
+type NodePoolUpdateDrainOptions struct {
+	Timeout int `mapstructure:"timeout"`
+
+	FailOnError bool `mapstructure:"failOnError"`
+
+	PodSelector string `mapstructure:"podSelector"`
+}
+
+// NodePool encapsulates information about a cluster node pool.
+type NodePool struct {
+	Name         string            `mapstructure:"name"`
+	Labels       map[string]string `mapstructure:"labels"`
+	Size         int               `mapstructure:"size"`
+	Autoscaling  Autoscaling       `mapstructure:"autoscaling"`
+	VolumeSize   int               `mapstructure:"volumeSize"`
+	InstanceType string            `mapstructure:"instanceType"`
+	Image        string            `mapstructure:"image"`
+	SpotPrice    string            `mapstructure:"spotPrice"`
```

Comment:
**Note**: I have just merged an EKS side change adding subnet ID to this structure which was accidentally missing from here and also I'm currently working on a change probably adding a status and a status message strings to the EKS equivalent of this struct. Not sure whether you want to update this one accordingly, just wanted to let you know in case you would.

Reply:
I just added the SubnetID as well, probably in the future it will be added (but not in this stage).